### ISSUE TITLE
Follow imported PropType objects

### ISSUE THREAD

**Issue by kumar303** [yannickcr/eslint-plugin-react#817]
It's common to use the same property type across many components so I would like to import my type definition. Would it be possible to support this?

Here is an example. First, create a common file like `propTypes.js` with:

``` javascript
import { PropTypes } from 'react';
export const i18nType = PropTypes.shape({});
```

Next, add it to a component's property types:

``` javascript
import React from 'react';

import { i18nType } from './propTypes';

export default class Thing extends React.Component {
  static propTypes = {
    i18n: i18nType.isRequired,
  }

  render() {
    return (
      <div>
        <span>{this.props.i18n.gettext('Some Text')}</span>
      </div>
    );
  }
}
```

As is, this should trigger an error like:

```
error  'i18n.gettext' is missing in props validation              react/prop-types
```

but it does not. It doesn't trigger an error at all, probably because it's not following the import.

The comments on https://github.com/yannickcr/eslint-plugin-react/issues/476 mention that this is too hard for a linter but that case was about following an object spread. This case is simpler because it's just a single import.

**Comment by ljharb** [comment]
Linters typically do static analysis - following imports to other files is generally far beyond the scope of a linter. It doesn't matter how many imports - looking beyond the literal source code in that file is what's challenging.

**Comment by kumar303** [comment]
It would be challenging in CommonJS but ES6 imports were designed specifically to [support static analysis](http://calculist.org/blog/2012/06/29/static-module-resolution/).

There are a couple eslint examples for how to follow imports. Here is [one](https://github.com/benmosher/eslint-plugin-import).

**Comment by ljharb** [comment]
Yes, but ES modules don't exist yet, because the loader spec is not finished - everything transpiles to `require` one way or another.

Note how I said "typically" and "generally" - eslint-plugin-import is the only popular one I know of that does, and it only does it for one or two of its rules. I'm claiming that it's out of scope.

**Comment by ljharb** [comment]
Linting across files is very complex and tricky, and there's a reason that eslint-plugin-import, and type systems like typescript/flow, are the only things that do it. We won't be able to add that here.
